### PR TITLE
Dispatch Authentication.logout event.

### PR DESCRIPTION
**Pull request by robertpustulka**
It's mentioned in the doc but hasn't been triggered.

### CODE REVIEW

File: src/Controller/Component/AuthenticationComponent.php [hunk 1]
```diff
@@ -83,13 +83,11 @@ protected function _afterIdentify()
...
 
-        $event = new Event('Authentication.afterIdentify', $this->_registry->getController(), [
+        $this->dispatchEvent('Authentication.afterIdentify', [
             'provider' => $provider,
             'identity' => $this->getIdentity(),
             'service' => $this->_authentication
-        ]);
-
-        $this->eventManager()->dispatch($event);
+        ], $this->_registry->getController());
```
Comment (cleptric) [inline src/Controller/Component/AuthenticationComponent.php]: Could this just be `$this->getController()` instead?

Comment (robertpustulka) [inline src/Controller/Component/AuthenticationComponent.php]: @cleptric right, thanks!

**Comment by cleptric** [comment]
Should we also add a `implementedEvents` method?

**Comment by robertpustulka** [comment]
@cleptric it's already there in parent `Component`.

**Comment by mehov** [comment]
Hi everyone,

The documentation at [/authentication/1/en/authenticators.html#events](https://book.cakephp.org/authentication/1/en/authenticators.html#events) and [/authentication/2/en/authenticators.html#events](https://book.cakephp.org/authentication/2/en/authenticators.html#events) has no mention of this. 

I also found cakephp/authentication#37 from earlier in 2017, i.e. a few months before this merge, which talks about actually removing `Authentication.logout` (via e9817e04de95289a811ffa669e056d42a81c4164).

Please clarify the confusion. Thank you!

**Comment by othercorey** [comment]
Please do not comment on old PRs.

**Pull request merged**